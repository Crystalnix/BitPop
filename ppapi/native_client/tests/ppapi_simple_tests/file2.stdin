# To be used with
#    sel_universal_ppapi_replay_prolog.stdin
# define a bunch of variables for readability

# NOTE: this test exercises the request.SetStreamToFile(true) case
#       which does not result in a 200 return code
#       So the expected outcome of this tests is a failure.

set_variable MODULE i(4444)
set_variable INSTANCE i(5555)

# Tags used by pepper/srpc to marshall js parameters
set_variable JS_STR 4:5
set_variable JS_INT 4:2

# NOTE: the strings below must not contain unescaped spaces or commas

set_variable FILE_DATA1 C(6,hello\x20)
set_variable FILE_SIZE1 i(6)
set_variable FILE_DATA2 C(6,world!)
set_variable FILE_SIZE2 i(6)

set_variable PARAM_STR_URL R(24,${JS_STR},4:10,"myurl.html",6:0)
set_variable PARAM_STR_GET R(16,${JS_STR},4:3,"GET",5:0)
set_variable PARAM_NUM_0  R(8,${JS_INT},4:0)
set_variable PARAM_NUM_1  R(8,${JS_INT},4:1)
set_variable PARAM_STATUS R(8,4:3,4:200)

set_variable NUM_TAGS i(3)
set_variable TAGS C(23,chunk_size\000url\000to_file\000)
set_variable VALUES C(16,10\000myurl.html\0001\000)

set_variable ERROR R(48,${JS_STR},4:33,"5555: Error: Finish unexpected rv",7:0)

######################################################################

# These replays are too common to list them separately
# TODO(robertm): make the counts tight, eliminate PPB_Var ASAP

replay 1 PPB_GetInterface:s:i  s("PPB_Messaging;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_Core;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_Var;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_URLRequestInfo;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_URLResponseInfo;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_URLLoader;1.0") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_FileIO;1.0") * i(1)

######################################################################

echo
echo "*** INIT MODULE"
# the results contain a process id which is non-determinisitic
rpc PPP_InitializeModule hide-results i(0) ${MODULE} h(pepper_desc) s("${service_string}") * i(0) i(0)

echo
echo "*** Load URL"
replay 1 PPB_URLRequestInfo_Create:i:i ${INSTANCE} * i(30)
replay 1 PPB_URLLoader_Create:i:i  ${INSTANCE} * i(34)
replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(0) ${PARAM_STR_URL} * i(1)
replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(1) ${PARAM_STR_GET} * i(1)
replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(3) ${PARAM_NUM_1} * i(1)
replay 1 PPB_URLLoader_Open:iii:i i(34) i(30) i(1) * i(-1)
replay 1 PPB_Core_ReleaseResource:i: i(30) *
rpc PPP_Instance_DidCreate ${INSTANCE} ${NUM_TAGS} ${TAGS} ${VALUES} * i(0)

echo
echo "*** async Open Callback"
replay 1 PPB_URLLoader_GetResponseInfo:i:i i(34) * i(38)
replay 1 PPB_URLResponseInfo_GetProperty:ii:C i(38) i(3) * ${PARAM_STATUS}
replay 1 PPB_URLLoader_ReadResponseBody:iii:Ci i(34) i(10) i(2) * C(0,) i(-1)
replay 1 PPB_Core_ReleaseResource:i: i(38) *
replay 1 PPB_Messaging_PostMessage:iC: ${INSTANCE} ${ERROR} *
rpc RunCompletionCallback i(1) i(0) C(0,) *


######################################################################
## EPILOG
######################################################################
echo
echo "*** DONE"
rpc PPP_ShutdownModule *


