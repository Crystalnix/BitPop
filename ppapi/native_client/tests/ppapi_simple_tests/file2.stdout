
"*** REPLAY PROLOG"

"*** REGISTER REPLAY METHODS"

"*** CREATE UPCALL SERVICE"

"*** INIT MODULE"
rpc call intiated PPP_InitializeModule:iihs:ii
input 0:  i(0)
input 1:  i(4444)
input 2:  h(pepper_desc)
input 3:  @LENGTHY_SERVICE_STRING
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_Core;1.0")
output 0:  i(1)
rpc call complete PPP_InitializeModule:iihs:ii


"*** Load URL"
rpc call intiated PPP_Instance_DidCreate:iiCC:i
input 0:  i(5555)
input 1:  i(3)
input 2:  C(23,chunk_size\x00url\x00to_file\x00)
input 3:  C(16,10\x00myurl.html\x001\x00)
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_URLLoader;1.0")
output 0:  i(1)
replaying PPB_URLLoader_Create:i:i:
input 0:  i(5555)
output 0:  i(34)
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_URLRequestInfo;1.0")
output 0:  i(1)
replaying PPB_URLRequestInfo_Create:i:i:
input 0:  i(5555)
output 0:  i(30)
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_Var;1.0")
output 0:  i(1)
replaying PPB_URLRequestInfo_SetProperty:iiC:i:
input 0:  i(30)
input 1:  i(0)
input 2:  C(24,\x05\x00\x00\x00\n\x00\x00\x00myurl.html\x00\x00\x00\x00\x00\x00)
output 0:  i(1)
replaying PPB_URLRequestInfo_SetProperty:iiC:i:
input 0:  i(30)
input 1:  i(3)
input 2:  C(8,\x02\x00\x00\x00\x01\x00\x00\x00)
output 0:  i(1)
replaying PPB_URLLoader_Open:iii:i:
input 0:  i(34)
input 1:  i(30)
input 2:  i(1)
output 0:  i(-1)
replaying PPB_Core_ReleaseResource:i::
input 0:  i(30)
rpc call complete PPP_Instance_DidCreate:iiCC:i
output 0:  i(1)


"*** async Open Callback"
rpc call intiated RunCompletionCallback:iiC:
input 0:  i(1)
input 1:  i(0)
input 2:  C(0,)
replaying PPB_URLLoader_GetResponseInfo:i:i:
input 0:  i(34)
output 0:  i(38)
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_URLResponseInfo;1.0")
output 0:  i(1)
replaying PPB_URLResponseInfo_GetProperty:ii:C:
input 0:  i(38)
input 1:  i(3)
output 0:  C(8,\x03\x00\x00\x00\xc8\x00\x00\x00)
replaying PPB_GetInterface:s:i:
input 0:  s("PPB_Messaging;1.0")
output 0:  i(1)
replaying PPB_Messaging_PostMessage:iC::
input 0:  i(5555)
input 1:  C(48,\x05\x00\x00\x00!\x00\x00\x005555: Error: Finish unexpected rv\x00\x00\x00\x00\x00\x00\x00)
replaying PPB_Core_ReleaseResource:i::
input 0:  i(38)
rpc call complete RunCompletionCallback:iiC:


"*** DONE"
rpc call intiated PPP_ShutdownModule::
rpc call complete PPP_ShutdownModule::

